creator-node/src/fileManager.js

```diff
@@ -295,17 +297,36 @@ async function saveFileForMultihashToFS (serviceRegistry, logger, multihash, exp
     }
 
     // verify that the contents of the file match the file's cid
+    let fileData = null
+    let fileSize = null
+    try {
+      fileData = (await fs.readFile(expectedStoragePath)).toString().slice(0, 99)
+      fileSize = (await fs.stat(expectedStoragePath)).size
+    } catch (e) {
+      logger.warn('Could not get fs statistics for path', expectedStoragePath)
+    }
     try {
-      decisionTree.push({ stage: 'About to verify the file contents for the CID', vals: multihash, time: Date.now() })
+      decisionTree.push({ stage: 'About to verify the file contents for the CID', vals: multihash, time: Date.now(), fileData, fileSize })
+
+      // if the file is empty, there's likely something wrong so throw an error
+      // there is one case where an empty file could be valid, check for that CID explicitly
+      const fileIsEmpty = fileSize !== null && fileSize === 0
+      if (fileIsEmpty && multihash !== EMPTY_FILE_CID) {
+        throw new Error(`File has no content, content length is 0: ${multihash}`)
+      }
+
       const ipfsHashOnly = await ipfsAdd.ipfsAddNonImages(expectedStoragePath, { onlyHash: true, timeout: 10000 })
       if (multihash !== ipfsHashOnly) {
         decisionTree.push({ stage: `File contents don't match IPFS hash multihash`, vals: ipfsHashOnly, time: Date.now() })
         // delete this file because the next time we run sync and we see it on disk, we'll assume we have it and it's correct
-        await fs.unlink(expectedStoragePath)
         throw new Error(`File contents don't match IPFS hash multihash: ${multihash} result: ${ipfsHashOnly}`)
       }
       decisionTree.push({ stage: 'Successfully verified the file contents for the CID', vals: multihash, time: Date.now() })
     } catch (e) {
+      await removeFile(expectedStoragePath)
+      if (numRetries > 0) {
+        return saveFileForMultihashToFS(serviceRegistry, logger, multihash, expectedStoragePath, gatewaysToTry, fileNameForImage, trackId, numRetries - 1)
```

Comment:
i actually don't think we should do this. the stream fetching is already optimized so it won't call all gateways unless it can't find it anywhere else. plus the ipfs and fs stuff is pretty cheap especially since this uses onlyhash. i think we should leave this as is